src/Compiler/Service/TransparentCompiler.fs

```diff
@@ -386,13 +386,31 @@ type internal TransparentCompiler
         (assumeDotNetFramework: bool option)
         (projectSnapshot: ProjectSnapshot)
         =
+        let useFsiAuxLib = defaultArg useFsiAuxLib true
+        let useSdkRefs = defaultArg useSdkRefs true
+        let assumeDotNetFramework = defaultArg assumeDotNetFramework false
+
+        let key =
+            { new ICacheKey<string * ProjectIdentifier, string> with
```

Comment:
I think this looks reasonable. 

I think we might be able to go with just file name and project identifier for key. Because hopefully for that combo we won't need multiple different script closures at the same time.

And we should probably include `OtherOptions` in the version.

And maybe then we can just pass ProjectIdentifier and OtherOptions instead of the snapshot so we don't have to create a fake snapshot in order to get a real snapshot 😬 

Although there's another question what ProjectIdentifier should even be for a script file.

Reply:
This script also depends on the included scripts. So if one of the included scripts changes, then we probably need to invalidate this one as well.

Reply:
@auduchinok while this is a valid remark, I don't think that level of detection is currently present in the background compiler. I would assume the options stamp currently is the only way to trigger cache invalidation of the load closure. Since none of the `FSharpProjectOptions` would have changed. when an included script changed.

The stamp should be in the new cache key as well.

Reply:
> Although there's another question what ProjectIdentifier should even be for a script file.

Yes, currently that is just script name + `.fsproj`. Not sure if a better name is possible here. As all we have is the entry script file.

Reply:
Yeah how do we deal with included scripts changing now? It could be done but maybe we'd need 2-phase process? Not really part of this PR but maybe something we need to look at in the future.

Reply:
Hmm, I don't think we do that in VS. Looking at the 2 usages of Stamp there it's either version of the file or project, both kinda redundant (from information point of view, in Options it's used for performance reasons I think) and won't change if included scripts change. And eventually I'd prefer to get rid of Stamp. But anyway we can deal with that later.

Reply:
Yes, I'm not really expecting anyone to actually do that, it seems like the only way to invalidate the cache in the old background compiler.

Reply:
> Yeah how do we deal with included scripts changing now?

IIRC this is fully recalculated every time script options are requested. Then, when a newer (and different) options are available, we use it instead of the ones that we've cached previously. I might be wrong, though.